internal/bminventory/inventory.go

```diff
@@ -4667,9 +4667,15 @@ func (b *bareMetalInventory) GetInfraEnvByKubeKey(key types.NamespacedName) (*co
 }
 
 func (b *bareMetalInventory) V2RegisterHost(ctx context.Context, params installer.V2RegisterHostParams) middleware.Responder {
+	var clusterID strfmt.UUID
+	var err error
+
 	log := logutil.FromContext(ctx, b.log)
 	log.Infof("Register host: %+v", params)
 
+	if clusterID, err = b.getClusterIDFromHost(*params.NewHostParams.HostID, params.InfraEnvID); err != nil {
```

Comment:
how will it work? the host is not registered yet, you need to check the infra-env here not the hosts table

Reply:
`getBoundCluster` is looking first at the infra env 
`getClusterIDFromHost` is looking at the hosts table
what did i miss?

Reply:
Changed to make it clearer.
Anyway, after the change getClusterIDFromHost will be called only if infra-env doesn't have cluster-id.
Before the getClusterIDFromHost  was called, but it was used only if infra-env doesn't have cluster-id (but it is irrelevant now).